### DISCUSSION TITLE
[@types/shopify-buy] Wrong types

### DISCUSSION THREAD

**Discussion by mlbonniec** [DefinitelyTyped/DefinitelyTyped#60911] [Issues with a @types package]
Hi ^^

I'm currently using the `shopify-buy` module with `@types/shopify-buy` declaration module, but I'm facing an issue, because all the types are wrong 🙁

### Demo
For example, I'm getting all my collections with their products like this.
The first issue is that some types are currently not ended, there's some 'Todo'. https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/shopify-buy/index.d.ts#L52-L59
```ts
const collections: Collection[] = await $client.collection.fetchAllWithProducts();
```

The second issue is that the `Collection` type doesn't match with the result ⤵️
```
// My console output after console.log(collections[0]);
GraphModel {
  id: [Getter],
  handle: [Getter],
  description: [Getter],
  descriptionHtml: [Getter],
  updatedAt: [Getter],
  title: [Getter],
  image: [Getter],
  products: [Getter],
  refetchQuery: [Getter],
  type: [Getter],
  nextPageQueryAndPath: [Function (anonymous)],
  hasNextPage: Scalar { value: true },
  hasPreviousPage: false,
  variableValues: { first: 20, productsFirst: 20 }
}
```
You can see here (https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/shopify-buy/index.d.ts#L291-L304) that, for example, the `products` field is missing, and property doesn't respect the same casing convention (snake case vs camel case).

I guess no, but is it normal? Or it's the declaration for an older Shopify API version?

[a comment by an automated account is not shown]

**Comment by straiforos** [comment]
@mlbonniec re: Old Shopify version; 
Likely some aspects are out of date I believe Shopify JS SDK is now at 2.17 yet those type errors you point out may have been an issue longer than that.

I helped update from a beta version to v1 with some errors here and there. Over the course of time, @MaciekBaron and @chrisworman-pela were the last two to bring it up to date with camel case updates etc. They likely can speak better to those inquiries. 

re is this normal? 
yes, this being a community drive type set and not supported by Shopify themselves it often has some minor flaws or out-of-date issues. Which is why when I went to use it I needed to do a bunch of learning. It was a good experience, and the library is easy to update if you so choose to contribute those findings.

**Comment by martin-badin** [comment]
Hi, I have tried to refactor those types. Please could you review it?
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/63339
